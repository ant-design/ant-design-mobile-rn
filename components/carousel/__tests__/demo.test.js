import rnDemoTest from '../../../tests/shared/demoTest'

rnDemoTest('carousel')
