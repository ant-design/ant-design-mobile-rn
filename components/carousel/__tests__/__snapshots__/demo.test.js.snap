// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/carousel/demo/basic.tsx correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <Text>
      horizontal
    </Text>
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <RNCViewPager
        initialPage={2}
        onMoveShouldSetResponderCapture={[Function]}
        onPageScroll={[Function]}
        onPageScrollStateChanged={[Function]}
        onPageSelected={[Function]}
        orientation="horizontal"
        showPageIndicator={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "height": 150,
            "width": "100%",
          }
        }
        transitionStyle="scroll"
      >
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "red",
                },
              ]
            }
          >
            <Text>
              Carousel 1
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "blue",
                },
              ]
            }
          >
            <Text>
              Carousel 2
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "yellow",
                },
              ]
            }
          >
            <Text>
              Carousel 3
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "aqua",
                },
              ]
            }
          >
            <Text>
              Carousel 4
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "fuchsia",
                },
              ]
            }
          >
            <Text>
              Carousel 5
            </Text>
          </View>
        </View>
      </RNCViewPager>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "position": "absolute",
            },
            Object {
              "bottom": 10,
              "left": 0,
              "right": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                Object {
                  "backgroundColor": "#888",
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
        </View>
      </View>
    </View>
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Go to 0
        </Text>
      </View>
    </>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <Text>
      vertical
    </Text>
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <RNCViewPager
        initialPage={1}
        onMoveShouldSetResponderCapture={[Function]}
        onPageScroll={[Function]}
        onPageScrollStateChanged={[Function]}
        onPageSelected={[Function]}
        orientation="vertical"
        showPageIndicator={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "height": 150,
            "width": "100%",
          }
        }
        transitionStyle="scroll"
      >
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "red",
                },
              ]
            }
          >
            <Text>
              Carousel 1
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "blue",
                },
              ]
            }
          >
            <Text>
              Carousel 2
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "yellow",
                },
              ]
            }
          >
            <Text>
              Carousel 3
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "aqua",
                },
              ]
            }
          >
            <Text>
              Carousel 4
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              Object {
                "bottom": 0,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "height": 150,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "fuchsia",
                },
              ]
            }
          >
            <Text>
              Carousel 5
            </Text>
          </View>
        </View>
      </RNCViewPager>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "position": "absolute",
            },
            Object {
              "bottom": 0,
              "right": 10,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                Object {
                  "backgroundColor": "#888",
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#cccccc",
                  "borderRadius": 8,
                  "height": 8,
                  "width": 8,
                },
                Object {
                  "marginHorizontal": 2.5,
                  "marginVertical": 3,
                },
                Object {},
                false,
                false,
              ]
            }
          />
        </View>
      </View>
    </View>
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Toggle autoplay 
          true
        </Text>
      </View>
    </>
  </View>
</View>
`;
